### PR TITLE
[datev2](function) make function nullable DEPEND_ON_ARGUMENT

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Set all functions' nullable mode for `datev2` to `DEPEND_ON_ARGUMENT`

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.51 seconds
 load time: 524 seconds
 storage size: 17122962414 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230128132528_clickbench_pr_85693.html

File: be/src/vec/functions/function_date_or_datetime_computation.h [hunk 1]
```diff
@@ -303,6 +305,21 @@ struct DateTimeOp {
...
+                              PaddedPODArray<ToType>& vec_to) {
+        size_t size = vec_from0.size();
+        vec_to.resize(size);
+
+        bool valid = true;
+        for (size_t i = 0; i < size; ++i) {
+            // here reinterpret_cast is used to convert uint8& to bool&,
+            // otherwise it will be implicitly converted to bool, causing the rvalue to fail to match the lvalue.
+            // the same goes for the following.
+            vec_to[i] = Transform::execute(vec_from0[i], vec_from1[i], valid);
```
Comment (HappenLee) [inline be/src/vec/functions/function_date_or_datetime_computation.h:319]: check the assembly code to make sure the function can SIMD ？

File: be/src/vec/functions/function_date_or_datetime_to_something.h [hunk 2]
```diff
@@ -82,11 +83,21 @@ class FunctionDateOrDateTimeToSomething : public IFunction {
                     get_name(), arguments.size());
         }
 
-        return make_nullable(std::make_shared<ToDataType>());
+        bool is_nullable = false;
+        bool is_datev2 = false;
+        for (auto it : arguments) {
+            is_nullable = is_nullable || it.type->is_nullable();
```
Comment (HappenLee) [inline be/src/vec/functions/function_date_or_datetime_to_something.h:89]:
```
        bool is_nullable = false;
        bool is_datev2 = false;
        for (auto it : arguments) {
            is_nullable = is_nullable || it.type->is_nullable();
            is_datev2 = is_datev2 || WhichDataType(remove_nullable(it.type)).is_date_v2() ||
                        WhichDataType(remove_nullable(it.type)).is_date_time_v2();
        }

        return is_nullable || !is_datev2 ? make_nullable(std::make_shared<ToDataType>())
                                         : std::make_shared<ToDataType>();
```
the code better simple to a template function?

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**